### PR TITLE
chore:adding skills back to infomon module

### PR DESCRIPTION
**Pull request by OSXLich-Doug**
Matching the Stats migration, this brings Skills back to the table for the new infomon release

### CODE REVIEW

File: lib/infomon/parser.rb [hunk 1]
```diff
@@ -86,10 +86,11 @@ def self.parse(line)
           Infomon.set("psm.%s" % match[:command], match[:ranks].to_i)
           :ok
         when Pattern::Skill
-          # todo: is there a need for ranks?
-          # todo: change Elemental Lore - Air to elair (and others)?
           match = Regexp.last_match
-          Infomon.set("skill.%s" % match[:name], match[:bonus].to_i)
+          Infomon.set("skill.%s" % match[:name].gsub(' ', '_').gsub('_-_', '').gsub('-', '_'),
```
Comment (ondreian) [inline lib/infomon/parser.rb:90]: I think this should be normalized in the `_key` method, same as other PR

File: lib/stats/skills.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      bonus
+    when String, Symbol
+      Infomon.get("skill.%s.bonus" % ranks)
+    else
+      echo "You're trying to move the cheese!"
+    end
+  end
+
+  @@skills = %i(two_weapon_combat armor_use shield_use combat_maneuvers edged_weapons blunt_weapons two_handed_weapons ranged_weapons thrown_weapons polearm_weapons brawling ambush multi_opponent_combat physical_fitness dodging arcane_symbols magic_item_use spell_aiming harness_power elemental_mana_control mental_mana_control spirit_mana_control elemental_lore_air elemental_lore_earth elemental_lore_fire elemental_lore_water spiritual_lore_blessings spiritual_lore_religion spiritual_lore_summoning sorcerous_lore_demonology sorcerous_lore_necromancy mental_lore_divination mental_lore_manipulation mental_lore_telepathy mental_lore_transference mental_lore_transformation survival disarming_traps picking_locks stalking_and_hiding perception climbing swimming first_aid trading pickpocketing)
+  # todo: lich up through 5.6.2 returns rank as integer - should we extend to include bonus?
```
Comment (ondreian) [inline lib/stats/skills.rb:38]: we should do `<skill>_bonus` imo as this is more "Rubyist"
`trading_bonus`
`two_weapon_combat_bonus`

**Review by ondreian — CHANGES_REQUESTED** [review]
looking very good just a few tweaks needed

File: spec/infomon_spec.rb [hunk 3]
```diff
@@ -115,12 +115,12 @@ def self.game
         Infomon::Parser.parse(line).eql?(:ok) or fail("did not parse:\n%s" % line)
       }
 
-      expect(Infomon.get("stat.fame")).to eq(4_804_958)
-      expect(Infomon.get("stat.fxp_current")).to eq(1_350)
-      expect(Infomon.get("stat.fxp_max")).to eq(1_010)
-      expect(Infomon.get("stat.ascension_experience")).to eq(4_170_132)
-      expect(Infomon.get("stat.total_experience")).to eq(41_307_131)
-      expect(Infomon.get("stat.long_term_experience")).to eq(26_266)
+      expect(Infomon.get("stat.fame")).to eq(4804958)
```
Comment (ondreian) [inline spec/infomon_spec.rb:118]: why did you change this? `_` numbers are much more readable

**Review by ondreian — CHANGES_REQUESTED** [review]

File: spec/infomon_spec.rb (hunk 3 above)
Comment (OSXLich-Doug) [reply spec/infomon_spec.rb:118]: I made two mistakes.  I corrected one.  I'll change that back now to fix the second.

**Review by ondreian — CHANGES_REQUESTED** [review]

File: lib/infomon/infomon.rb [hunk 4]
```diff
@@ -56,6 +56,7 @@ def self.setup!
   end
 
   def self._key(key)
+    key.gsub!(' ', '_').gsub!('_-_', '').gsub!('-', '_') if key =~ /\s|-/
```
[the remark this answers is not in the document]
Comment (OSXLich-Doug) [reply lib/infomon/infomon.rb]: Done

**Review by ondreian — APPROVED** [review]
